src/AbstractSQLCompiler.ts

```diff
@@ -448,6 +451,14 @@ export interface Trigger {
 	level: 'ROW' | 'STATEMENT';
 	when: 'BEFORE' | 'AFTER' | 'INSTEAD OF';
 }
+export interface Index {
+	type: string;
+	fields: string[];
+	name?: string;
+	description?: string;
+	nulls?: string;
```

Comment:
I think this should probably be a boolean, eg
```suggestion
	distinctNulls?: boolean;
```
or at the very least specific on the strings, eg
```suggestion
	nulls?: 'DISTINCT' | 'NOT DISTINCT';
```